basket/base/middleware.py

```diff
@@ -0,0 +1,58 @@
+import contextlib
+import inspect
+import time
+
+from django.conf import settings
+from django.http import Http404
+from django.utils.deprecation import MiddlewareMixin
+
+from basket import metrics
+
+
+class HostnameMiddleware(MiddlewareMixin):
+    """Add header with k8s cluster / pod details for debugging."""
+
+    def process_response(self, request, response):
+        response["X-Backend-Server"] = "/".join(filter(None, [getattr(settings, x) for x in ["CLUSTER_NAME", "K8S_NAMESPACE", "K8S_POD_NAME"]]))
+        return response
+
+
+class MetricsStatusMiddleware(MiddlewareMixin):
+    """Send status code counts to statsd"""
+
+    def process_response(self, request, response):
+        metrics.incr(f"response.{response.status_code}")
+        return response
+
+    def process_exception(self, request, exception):
+        if not isinstance(exception, Http404):
+            metrics.incr("response.500")
+
+
+class MetricsRequestTimingMiddleware(MiddlewareMixin):
+    """Send request timing to statsd"""
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        if inspect.isfunction(view_func):
+            view = view_func
+        else:
+            view = view.__class__
+
+        with contextlib.suppress(AttributeError):
+            request._start_time = time.time()
+            request._view_module = view.__module__
+            request._view_name = view.__name__
+
+    def _record_timing(self, request):
+        if hasattr(request, "_start_time"):
+            view_time = int((time.time() - request._start_time) * 1000)
+            metrics.timing(f"view.{request._view_module}.{request._view_name}.{request.method}", view_time)
+            metrics.timing(f"view.{request._view_module}.{request.method}", view_time)
```

Comment:
Above, you're suppressing a possible `AttributeError`, but that means that `request._view_module` and `request._view_name` may not exist, which will (I'm pretty sure) trigger an attribute error in these two lines.

I think instead, I'd drop suppressing the `AttributeError` and change lines 43 and 44 to something like:

```python
request._view_module = getattr(view, '__module__', 'nomodule')
request._view_name = getattr(view, '__name__', 'noname')
```